obsreport/obsreport_test.go

```diff
@@ -57,16 +54,14 @@ type testParams struct {
 }
 
 func TestReceiveTraceDataOp(t *testing.T) {
-	doneFn, err := obsreporttest.SetupRecordedMetricsTest()
+	set, doneFn, err := obsreporttest.SetupRecordedMetricsTest()
 	require.NoError(t, err)
 	defer doneFn()
 
-	sr := new(tracetest.SpanRecorder)
-	tp := sdktrace.NewTracerProvider(sdktrace.WithSpanProcessor(sr))
-	otel.SetTracerProvider(tp)
+	otel.SetTracerProvider(set.TracerProvider)
```

Comment:
I think we can get rid of this since we now pass the provider to the grpc/http configs

Reply:
I left this here as the receiver still uses the global tracer provider:https://github.com/open-telemetry/opentelemetry-collector/blob/b39f52bfe67b03d0ff7b3601d3b48bf1ce41c8ac/obsreport/obsreport_receiver.go#L63